libcontainer/configs/config.go

```diff
@@ -205,15 +205,55 @@ type Config struct {
 type Hooks struct {
 	// Prestart commands are executed after the container namespaces are created,
 	// but before the user supplied command is executed from init.
+	// Note: This hook is now deprecated
+	// Prestart commands are called in the Runtime namespace.
 	Prestart []Hook
 
+	// CreateRuntime commands MUST be called as part of the create operation after
+	// the runtime environment has been created but before the pivot_root has been executed.
+	// CreateRuntime is called immediately after the deprecated Prestart hook.
+	// CreateRuntime commands are called in the Runtime Namespace.
+	CreateRuntime []Hook
+
+	// CreateContainer commands MUST be called as part of the create operation after
+	// the runtime environment has been created but before the pivot_root has been executed.
+	// CreateContainer commands are called in the Container namespace.
+	CreateContainer []Hook
+
+	// StartContainer commands MUST be called as part of the start operation and before
+	// the container process is started.
+	// StartContainer commands are called in the Container namespace.
+	StartContainer []Hook
+
 	// Poststart commands are executed after the container init process starts.
+	// Poststart commands are called in the Runtime Namespace.
 	Poststart []Hook
 
 	// Poststop commands are executed after the container init process exits.
+	// Poststop commands are called in the Runtime Namespace.
 	Poststop []Hook
 }
 
+type HookName int
+
+const (
+	Prestart HookName = iota
+	CreateRuntime
+	CreateContainer
+	StartContainer
+	Poststart
+	Poststop
+)
+
+var HookToName = map[HookName]string{
+	Prestart:        "prestart",
+	CreateRuntime:   "createRuntime",
+	CreateContainer: "createContainer",
+	StartContainer:  "startContainer",
+	Poststart:       "poststart",
+	Poststop:        "poststop",
+}
```

Comment:
It'd be simpler to just make `type HookName string` and define the constants as strings.